### ISSUE TITLE
Better handling of `<aside>` content

### ISSUE THREAD

**Issue by danburzo** [mozilla/readability#487]
As per [this article](http://html5doctor.com/aside-revisited/), an `<aside>` inside an `<article>` should be treated as part of the content. In @andreskrey's [Pull request](https://github.com/mozilla/readability/pull/431) all `aside` elements are removed, and articles [such as this](https://hpbn.co/primer-on-latency-and-bandwidth/) get useful content removed. 

Unfortunately, it seems Engadget uses asides in the `<article>` tag for content unrelated to the article... so I'm not sure how a good solution would look like.

**Comment by andreskrey** [comment]
> Unfortunately, it seems Engadget uses asides in the `<article>` tag for content unrelated to the article... so I'm not sure how a good solution would look like.

Don't you mean `<aside>`?

**Comment by andreskrey** [comment]
Can you provide an example that's getting parsed incorrectly?

**Comment by danburzo** [comment]
Yes, this article: https://hpbn.co/primer-on-latency-and-bandwidth/ uses `<aside>` as intended, and the blocks get stripped. Maybe we can look into whether the `<aside>` has meaningful content?

**Comment by AlbertSPedersen** [comment]
Bumping this. Cloudflare's documentation side makes heavy use of `<aside>` fact/info boxes. This is arguably part of the main content and should be included in reader view, but it currently is not.

![image](https://github.com/mozilla/readability/assets/40433209/939e4c0b-e20e-4bac-9dee-30040315cd81)

**Comment by hollisticated-horse** [comment]
Hello all, what's the ETA/state of `<aside>` support here ? 

We are also having issues with aside elements @[Readeck](https://codeberg.org/readeck/readeck/issues/325#issuecomment-2474164) which uses Readability.js for content extraction.

**Comment by danburzo** [comment]
The HTML Standard has overloaded [aside elements](https://html.spec.whatwg.org/multipage/sections.html#the-aside-element) to signify everything only tangentially related to the article, including navigation (related articles), advertising, and pull quotes. This has created a situation where some of these uses should be stripped from the content, and others should be kept. Probably the only realistic path forward is providing a user option for handling asides.

**Comment by hollisticated-horse** [comment]
> The HTML Standard has overloaded [aside elements](https://html.spec.whatwg.org/multipage/sections.html#the-aside-element) to signify everything only tangentially related to the article, including navigation (related articles), advertising, and pull quotes. 

Could there be a way to heuristically categorize asides based on it's content? 

> This has created a situation where some of these uses should be stripped from the content, and others should be kept. Probably the only realistic path forward is providing a user option for handling asides.
Can you point me to code I can take a look at that's relevant to this option for showing/hiding aside content?

**Comment by danburzo** [comment]
At first glance asides are simply removed with: 

- https://github.com/mozilla/readability/blob/118f01538e167218bd86ffd493bd3466aec4870a/Readability.js#L798

**Comment by gijsk** [comment]
#431 added this; as noted already various high-profile sites like Engadget, the Verge, possibly the NY Times (hard to tell with the paywall), and probably many others, use `aside` for content not related to the main article. I'd be happy to see a PR but it would need to not regress popular sites like that (ideally with both the existing testcase and a more recent one), and for that reason probably isn't as simple as just removing the line mentioned earlier.

Safari's reader mode code isn't public (never mind compatibly licensed), so we cannot really "follow" it.
